File: Codigo/apps/mobile/lib/features/checkin/domain/checkin.dart
class CheckinAuthor {
  final String? id;
  final String? name;
  final String? image;

  CheckinAuthor({this.id, this.name, this.image});

  factory CheckinAuthor.fromJson(Map<String, dynamic>? json) {
    if (json == null) return CheckinAuthor();
    return CheckinAuthor(
      id: json['id'] as String?,
      name: json['name'] as String?,
      image: json['image'] as String?,
    );
  }
}

class Checkin {
  final String id;
  final String title;
  final String? description;
  final String? image;
  final String? summaryAi;
  final int points;
  final DateTime createdAt;
  final CheckinAuthor author;

  Checkin({
    required this.id,
    required this.title,
    this.description,
    this.image,
    this.summaryAi,
    required this.points,
    required this.createdAt,
    required this.author,
  });

  factory Checkin.fromJson(Map<String, dynamic> json) {
    return Checkin(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      description: json['description'] as String?,
      image: json['image'] as String?,
      summaryAi: json['summary_ai'] as String?,
      points: json['points'] is int ? json['points'] as int : int.tryParse('${json['points']}') ?? 0,
      createdAt: DateTime.tryParse(json['createdAt']?.toString() ?? '') ?? DateTime.now(),
      author: CheckinAuthor.fromJson(json['author'] as Map<String, dynamic>?),
    );
  }
}

File: Codigo/apps/backend/src/main/java/dev/coderats/backend/domain/Checkin.java
package dev.coderats.backend.domain;

import java.time.OffsetDateTime;
import java.time.ZoneOffset;
import java.util.UUID;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;

@Entity
@Table(name = "checkins")
public class Checkin {
    
    @Id
    @GeneratedValue
    private UUID id;
    
    @Column(name = "user_id", nullable = false)
    private UUID userId;
    
    @Column(name = "group_id", nullable = false)
    private UUID groupId;
    
    @Column(nullable = false)
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(columnDefinition = "TEXT")
    private String image;
    
    @Column(name = "summary_ai", columnDefinition = "TEXT")
    private String summaryAi;
    
    @Column(nullable = false)
    private int points = 0;
    
    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt;
    
    @Column(name = "updated_at", nullable = false)
    private OffsetDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private OffsetDateTime deletedAt;
    
    public Checkin() {}
    
    public Checkin(UUID userId, UUID groupId, String title, String description, String image, String summaryAi, int points) {
        this.userId = userId;
        this.groupId = groupId;
        this.title = title;
        this.description = description;
        this.image = image;
        this.summaryAi = summaryAi;
        this.points = points;
    }
    
    @PrePersist
    void prePersist() {
        createdAt = updatedAt = OffsetDateTime.now(ZoneOffset.UTC);
    }
    
    @PreUpdate
    void preUpdate() {
        updatedAt = OffsetDateTime.now(ZoneOffset.UTC);
    }
    
    // Getters e Setters
    public UUID getId() { return id; }
    public void setId(UUID id) { this.id = id; }
    
    public UUID getUserId() { return userId; }
    public void setUserId(UUID userId) { this.userId = userId; }
    
    public UUID getGroupId() { return groupId; }
    public void setGroupId(UUID groupId) { this.groupId = groupId; }
    
    public String getTitle() { return title; }
    public void setTitle(String title) { this.title = title; }
    
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    
    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }
    
    public String getSummaryAi() { return summaryAi; }
    public void setSummaryAi(String summaryAi) { this.summaryAi = summaryAi; }
    
    public int getPoints() { return points; }
    public void setPoints(int points) { this.points = points; }
    
    public OffsetDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(OffsetDateTime createdAt) { this.createdAt = createdAt; }
    
    public OffsetDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(OffsetDateTime updatedAt) { this.updatedAt = updatedAt; }
    
    public OffsetDateTime getDeletedAt() { return deletedAt; }
    public void setDeletedAt(OffsetDateTime deletedAt) { this.deletedAt = deletedAt; }
}
